tests/20-rdfc-verify.js

```diff
@@ -1,84 +1,183 @@
-        credential: credentials.verify.document,
-        suite: 'ecdsa-rdfc-2019',
-        keyTypes
-      });
-    });
-    for(const [name, {endpoints: verifiers}] of match) {
+    setupReportableTestSuite(this, 'Verifier');
+
+    /**
+     * We build a dynamic test suite based on the implementations that support
+     * the `ecdsa-rdfc-2019` suite.
+     *
+     * If the target implementations do not support a specific test we will
+     * skip the test but still keep it defined so that this is obvious in the
+     * results.
+     *
+     * Note that a single implementation may define multiple verifiers, which
+     * in turn may support multiple key types.
+     *
+     * We want a test suite for each permutation for every key type we require.
+    */
+
+    for(const [name, {endpoints: verifiers}] of targetImplementations) {
       for(const verifier of verifiers) {
         const {
           supportedEcdsaKeyTypes: verifierKeyTypes
         } = verifier.settings;
-        const keyTypes = verifierKeyTypes.join(', ');
-        // add implementer name and keyTypes to test report
-        this.implemented.push(`${name}: ${keyTypes}`);
-        describe(`${name}: ${keyTypes}`, function() {
-          let supportedVectors = [];
-          before(function() {
-            // filter the test data to only include VC signed with
-            // keyTypes the verifier supports
-            supportedVectors = verifierKeyTypes.map(
-              (keyType = '') => testVectors.get(keyType.toUpperCase()));
-          });
-          // wrap the testApi config in an Implementation class
-          it('MUST verify a valid VC with an ecdsa-rdfc-2019 proof.',
-            async function() {
-              this.test.cell = {
-                columnId: `${name}: ${keyTypes}`, rowId: this.test.title
-              };
-              for(const vector of supportedVectors) {
-                await verificationSuccess({credential: vector, verifier});
-              }
-            });
-          it('If the "cryptosuite" field is not the string ' +
-            '"ecdsa-rdfc-2019" or "ecdsa-jcs-2019", an error MUST be ' +
-            'raised.', async function() {
-            this.test.cell = {
-              columnId: `${name}: ${keyTypes}`, rowId: this.test.title
-            };
-            for(const vector of supportedVectors) {
-              const credential = klona(vector);
-              //FIXME add invalid-cryptosuite as a valid cryptosuite name
-              //locally so the signature is correct, but the cryptosuite
-              //name is incorrect
-              credential.proof.cryptosuite = 'invalid-cryptosuite';
-              await verificationFail({credential, verifier});
-            }
+
+        // Important: iterates over required keytypes
+        // (vs those specified by the implementation).
+        for(const keyType of requiredSupportedKeyTypes) {
+          const supportsKeyType = verifierKeyTypes.includes(keyType);
```

Comment:
do remember that not all issuers support all keyTypes so throwing because an implementer doesn't support `p-384` is probably not a good idea. We should on the other hand throw if they have no supportedEcdsaKeyTypes.

Reply:
marking this as resolved as apparently we want to mark implementations that don't support both keyTypes as failed.